endpoints/setuid.go

```diff
@@ -164,8 +164,19 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 
 		setSiteCookie := siteCookieCheck(r.UserAgent())
 
+		// Priority Ejector Set Up
+		if !isSyncerPriority(syncer.Key(), cfg.UserSync.PriorityGroups, syncersByBidder) {
+			err = errors.New("syncer key " + syncer.Key() + " is not in priority groups")
+			w.WriteHeader(http.StatusBadRequest)
+			metricsEngine.RecordSetUid(metrics.SetUidBadRequest)
+			so.Errors = []error{err}
+			so.Status = http.StatusBadRequest
+			return
+		}
+		priorityEjector := &usersync.PriorityBidderEjector{PriorityGroups: cfg.UserSync.PriorityGroups, TieEjector: &usersync.OldestEjector{}}
```

Comment:
The check in line 168 implies that: if we made it all the way here, `cfg.UserSync.PriorityGroups` is not empty which makes lines 77 to 79 of `usersync/ejector.go` to be logically dead code that we can remove:
```
75   func getNonPriorityUids(uids map[string]UIDEntry, priorityGroups [][]string, syncersByBidder map[string]Syncer) map[string]UIDEntry {
76 -     // If no priority groups, then all keys in uids are non-priority
77 -     if len(priorityGroups) == 0 {
78 -         return uids
79 -     }
80  
81      // Create map of keys that are a priority
82      isPriority := make(map[string]bool)
83      for _, group := range priorityGroups {
84          for _, bidder := range group {
85              if bidderSyncer, ok := syncersByBidder[bidder]; ok {
usersync/ejector.go
```

Reply:
My comment below talking about my updates to how IsSyncerPriority() is used results in this length check still being needed here